lighthouse-core/gather/computed/computed-artifact.js

```diff
@@ -38,18 +38,21 @@ class ComputedArtifact {
 
   /**
    * Request a computed artifact, caching the result on the input artifact.
-   * @param {!OBject} artifact
+   * @param {!Object} artifact
+   * @param {!Object=} artifacts
    * @return {!Promise}
    */
-  request(artifact) {
+  request(artifact, artifacts) {
```

Comment:
the problem is that it breaks the cache and `requiredArtifacts` dependency checks, since the computed artifact could theoretically access any artifact it wanted at this point. In this case it's only accessing other computed artifacts, so it's fine, but we can't guarantee it.

I have a proposal I've been thinking about for a while (it also makes typing nicer for the artifacts object). I'll throw it together quickly and put it in a PR